client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -482,25 +484,42 @@ export default {
     },
 
     updateBlocks (passedBlock = undefined) {
-      this.editor.block.meta.tempID = this.editor.block.meta.tempID || this.generateUID()
-      const block = compose.PageBlockMaker(this.editor.block)
-      this.page.blocks = this.blocks
+      if (passedBlock && 'block' in passedBlock) {
+        const block = compose.PageBlockMaker(passedBlock.block)
+        block.meta.tempID = block.meta.tempID || this.generateUID()
```

Comment:
Dont see why you dont add this logic on the class itself like we discussed. Same for getID of block